test-sbt/shared/src/main/scala/zio/test/sbt/BaseTestTask.scala

```diff
@@ -26,30 +25,7 @@ abstract class BaseTestTask(
 
   protected def run(eventHandler: EventHandler) =
     for {
-      spec <- (args.testSearchTerms, args.tagSearchTerms) match {
-               case (Nil, Nil) => spec.run
-               case (testSearchTerms, Nil) =>
-                 spec.runner.run {
-                   spec.spec.filterLabels { label =>
-                     testSearchTerms.exists(term => label.contains(term))
-                   }.getOrElse(spec.spec)
-                 }
-               case (Nil, tagSearchTerms) =>
-                 spec.runner.run {
-                   spec.spec.filterTags { tag =>
-                     tagSearchTerms.exists(term => tag == term)
-                   }.getOrElse(spec.spec)
-                 }
-               case (testSearchTerms, tagSearchTerms) =>
-                 spec.runner.run {
-                   spec.spec.filterTags { tag =>
-                     testSearchTerms.exists(term => tag == term)
-                   }.flatMap(_.filterLabels { label =>
-                       tagSearchTerms.exists(term => label.contains(term))
-                     })
-                     .getOrElse(spec.spec)
-                 }
-             }
+      spec    <- specInstance.runner.run(FilteredSpec(specInstance.spec, args))
```

Comment:
Is there a difference between running the spec directly with `spec.run` or passing it to the `runner`?

Reply:
Whatever it is, if I run the (unfiltered) spec with `specInstance.runner.run`, it causes `StackTracesSpec` to fail, for some reason... The only difference, as far as I can tell, is this skips weaving of the aspects (which is currently just the timeout?). Unless there's something I'm missing, @adamgfraser?

Reply:
@hmemcpy I think that should be the only difference. `StackTracesSpec` can be a little finicky because it depends on the exact trace that is rendered so if you do something that causes an additional step to be included in the trace the test can fail.  There is a debug option in that test suite if you want to figure it out but I think you want to weave in the test aspects, right?

Reply:
@adamgfraser I think so... in the original code it seems, though, that if there's no filter, the entire spec is run with all aspects applied (1):
```
case (Nil, Nil) => spec.run    <-- (1)
               case (testSearchTerms, Nil) =>
                 spec.runner.run {  <-- (2)
                   spec.spec.filterLabels { label =>
                     testSearchTerms.exists(term => label.contains(term))
                   }.getOrElse(spec.spec)
                 }
                 ...
```

however, if there are filtered items, it runs it through the runner directly, bypassing the aspect weaving (2).

After my change, it always runs with aspects applied, regardless of whether it was filtered or not. All tests are passing, so I don't think that's a problem.

Reply:
@adamgfraser Wow, this actually was a bug! Filtered items did not have aspects applied to them. I inadvertently fixed it by always applying the aspects!

If you currently run e.g. `coreTestsJVM/testOnly *.StackTracesSpec -- -t foreachTrace` with the master branch, you'll get test failures!